<?xml version="1.0" encoding="UTF-8"?>
<launch>
 
    <node pkg="cones_perception" type="cone_detection" name="cone_detection" output="screen">
        <param name="cones_frame_id" type="str" value="cloud"/>
        <param name="cones_topic" type="str" value="/cones_cloud"/>
        <param name="classify_colors" type="bool" value="false"/>

        <param name="input_cloud_topic" type="str" value="/fsds/lidar/Lidar"/>
        <param name="distance_treshold_max" type="double" value="10.0"/>
        <param name="distance_treshold_min" type="double" value="1.0"/>
        <param name="level_threshold" type="double" value="-5"/>
        <param name="angle_threshold" type="double" value="160.0"/>
        <param name="min_cluster_size" type="int" value="2"/>
        <param name="max_cluster_size" type="int" value="500"/>
    </node>

    <!-- Rviz visualization-->
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find cones_perception)/rviz/for_simulation.rviz" />

     <!-- Static transforms between scanners and map frame -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="link_map_cloud" args="0 0 0 0 0 0 map cloud" />  
    <node pkg="tf2_ros" type="static_transform_publisher" name="link_cloud_lidar" args="0 0 0 0 0 0 cloud lidar" />

</launch>